packages/design-system/src/components/web-components/ds-choice/ds-choice.test.tsx

```diff
@@ -0,0 +1,266 @@
+    expect(input).toHaveClass('foo');
+  });
+
+  it('applies custom class to label', () => {
+    const { container } = renderChoice({ 'label-class-name': 'bar' });
+    const label = container.querySelector('label');
+    expect(label).toHaveClass('bar');
+  });
+
+  it('applies errorMessage to label', () => {
+    const errorMessage = "Hey! You can't do that!";
+    const { asFragment } = renderChoice({ 'error-message': errorMessage });
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('applies inverse className to input', () => {
+    renderChoice({ inversed: true });
+    const input = screen.getByRole('checkbox');
+    expect(input).toHaveClass('ds-c-choice--inverse');
+  });
+
+  it('applies small className to input', () => {
+    renderChoice({ size: 'small' });
+    const input = screen.getByRole('checkbox');
+    expect(input).toHaveClass('ds-c-choice--small');
+  });
+
+  it('applies additional classNames to root element', () => {
+    const { container } = renderChoice({ className: 'foo' });
+    const el = container.firstChild as HTMLElement;
+    expect(el).toHaveClass('foo');
+  });
+
+  it('accepts a string value in the input', () => {
+    renderChoice({ value: 'bar' });
+    const input = screen.getByRole('checkbox');
+    expect(input).toHaveAttribute('value', 'bar');
+  });
+
+  it('accepts a number value in the input', () => {
+    renderChoice({ value: 100 });
+    const input = screen.getByRole('checkbox');
+    expect(input).toHaveAttribute('value', '100');
+  });
+
+  it('accepts a custom id', () => {
+    renderChoice({ 'root-id': 'custom_id' });
+    const el = screen.getByRole('checkbox');
+    expect(el.id).toBe('custom_id');
+  });
+
+  it('generates a unique id', () => {
+    const props = {
+      name: 'presidents',
+      label: defaultAttrs.label,
+      type: defaultAttrs.type,
+    };
+    const { container } = render(
+      <div>
+        <ds-choice value="a" {...props} />
+        <ds-choice value="b" {...props} />
+      </div>
+    );
+
+    const idRegex = /choice--\d+/;
+    const labels = container.querySelectorAll('label');
+    const labelA = labels[0];
+    const labelB = labels[1];
+    const inputA = screen.getByDisplayValue('a');
+    const inputB = screen.getByDisplayValue('b');
+
+    // IDs should be unique!
+    expect(inputA.id).not.toBe(inputB.id);
+
+    // First Choice
+    expect(inputA.id).toMatch(idRegex);
+    expect(labelA.getAttribute('for')).toBe(inputA.id);
+
+    // Second choice
+    expect(inputB.id).toMatch(idRegex);
+    expect(labelB.getAttribute('for')).toBe(inputB.id);
+  });
+
+  it('fires a custom ds-change event', () => {
+    renderChoice();
+
+    const choiceRoot = document.querySelector('ds-choice');
+    const mockHandler = jest.fn();
+    choiceRoot.addEventListener('ds-change', mockHandler);
+
+    const input = screen.getByRole('checkbox');
+    userEvent.click(input);
+
+    expect(mockHandler).toHaveBeenCalledTimes(1);
```

Comment:
I wanted to assert here that `mockHandler.mock.lastCall[0].details.target.checked` was true, but the event object is kind of messed up. I'm wondering if it's a limitation with jsdom and custom events. It seems to work right in Storybook.